fix(samples): Update KFP version in sample to fix backslash recognization issue.

zijianjoy opened the pull request:
**Description of your changes:**
There was an issue at 1.3.0 where pipeline failed to recognize backslash. Upgrade KFP SDK to 1.4.0 fix the issues.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
<!--
   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.
-->
- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).
<!--
    **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update before release.
-->

zijianjoy commented:
See screenshot before fix:
<img width="1381" alt="Screen Shot 2021-02-16 at 2 33 35 PM" src="https://user-images.githubusercontent.com/37026441/108130206-24b7d500-7064-11eb-9b1a-c401f57a4ddc.png">

zijianjoy commented:
After fix:

<img width="1535" alt="Screen Shot 2021-02-16 at 2 39 16 PM" src="https://user-images.githubusercontent.com/37026441/108130598-c3443600-7064-11eb-85f4-82190e0622be.png">

Ark-kun commented:
Can you please try the latest SDK?
There was a bug which affected escape characters. It's now fixed.

(The bug was due to `echo` replacing escape characters. We had to replace it with `printf`.)

zijianjoy commented:
> Can you please try the latest SDK?
> There was a bug which affected escape characters. It's now fixed.
> 
> (The bug was due to `echo` replacing escape characters. We had to replace it with `printf`.)

Thank you so much Alexey! I have changed the PR to restrict user to use the latest SDK version so to fix the issue.

Ark-kun commented:
Thank you for improving teh sample.
/lgtm
/approve
